pgo/cmd/auth.go

```diff
@@ -242,16 +261,40 @@ func GetCredentials() {
 		os.Exit(2)
 	}
 
-	log.Debug("setting up httpclient with TLS")
-	httpclient = &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs:            caCertPool,
-				InsecureSkipVerify: true,
-				Certificates:       []tls.Certificate{cert},
-				MinVersion:         tls.VersionTLS11,
-			},
+	// create a Transport object for use by the HTTP client
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs:            caCertPool,
+			InsecureSkipVerify: true,
+			Certificates:       []tls.Certificate{cert},
+			MinVersion:         tls.VersionTLS11,
 		},
 	}
-
+	return tr
 }
+
+func SetHttpClient(tr *http.Transport){
+	// If TLS is disabled, establish connection with
+	// a 60 second timeout.
+	if tr == nil {
+		log.Debug("setting up httpclient without TLS")
+		httpclient = &http.Client{
+			Timeout:    httpTimeout * time.Second,
+			}
+	} else {
+	// If TLS is not disabled, gather necessary information
+	// to establish a secure connection with a 60 second timeout.
+	log.Debug("setting up httpclient with TLS")
+		httpclient = &http.Client{
+			Timeout:		httpTimeout * time.Second,
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs:            caCertPool,
+					InsecureSkipVerify: true,
+					Certificates:       []tls.Certificate{cert},
+					MinVersion:         tls.VersionTLS11,
+				},
+			},
+		}
+	}
```

Comment:
A few things:

1. Please document the function top, using `godoc` format
1. I'd recommend using a more descriptive variable name, e.g. `httpTransport` instead of `tr` (I'll refer to it as `tr` in the rest of the comments here for simplicity)
1. I'd recommend starting with the `tr != nil` case first, that seems to be "go" way of doing things. It will also be more concise.
1. This does not appear to work, the `tr` parameter passed in is not used in the `tr != nil` case. Example that is not gofmt'd:

```go
// If TLS is enabled, use the transport information already provided to
// establish a secure connection provided with the timeout specified
// in "httpTimeout"
//
// Otherwise, setup a HTTP connection with the timeout specified
// in "httpTimeout"
if tr != nil {
    httpclient = &http.Client{
        Timeout: httpTimeout * time.Second,
        Transport: tr
    }
} else {
    httpclient = &http.Client{
        Timeout:    httpTimeout * time.Second,
    }
}
```
(with appropriate debugging lines)

Reply:
Updated for all items as requested.